core/src/main/java/io/aiven/klaw/helpers/db/rdbms/SelectDataJdbc.java

```diff
@@ -376,66 +384,165 @@ public List<Acl> getPrefixedAclsSOT(String env, int tenantId) {
 
     if (allReqs) { // approvers
-
+      // Team Name is null here as it is the team who created the requested
+      // Description includes the teamId as that is how Claim topics know who is the owning team.
       List<TopicRequest> claimTopicReqs =
-          topicRequestsRepo.findAllByTopictypeAndTenantId(
-              RequestOperationType.CLAIM.value, tenantId);
-      topicRequests =
-          topicRequests.stream()
+          Lists.newArrayList(
+              findTopicRequestsByExample(
+                  RequestOperationType.CLAIM.value,
+                  null,
+                  env,
+                  status,
+                  tenantId,
+                  String.valueOf(teamSelected)));
+
+      topicRequestListSub =
+          Lists.newArrayList(findTopicRequestsByExample(null, teamId, env, status, tenantId, null));
+
+      // Only execute just before adding the separate claim list as this will make sure only the
+      // claim topics this team is able to approve will be returned.
+      topicRequestListSub =
+          topicRequestListSub.stream()
               .filter(
                   topicRequest ->
                       !RequestOperationType.CLAIM.value.equals(topicRequest.getTopictype()))
               .collect(Collectors.toList());
-      claimTopicReqs =
-          claimTopicReqs.stream()
-              .filter(
-                  claimTopicReq ->
-                      Objects.equals(claimTopicReq.getDescription(), "" + teamSelected))
-              .collect(Collectors.toList());
-
-      topicRequests.addAll(claimTopicReqs);
 
+      topicRequestListSub.addAll(claimTopicReqs);
       // remove users own requests to approve/show in the list
-      topicRequests =
-          topicRequests.stream()
+      // Placed here as it should only apply for approvers.
+      topicRequestListSub =
+          topicRequestListSub.stream()
               .filter(topicRequest -> !topicRequest.getRequestor().equals(requestor))
               .collect(Collectors.toList());
-    }
+    } else {
+      if (showRequestsOfAllTeams) {
+        topicRequestListSub =
+            Lists.newArrayList(findTopicRequestsByExample(null, null, null, null, tenantId, null));
+      } else {
 
-    if (!"all".equals(status)) {
-      topicRequests =
-          topicRequests.stream()
-              .filter(topicRequest -> topicRequest.getTopicstatus().equals(status))
-              .collect(Collectors.toList());
+        topicRequestListSub =
+            Lists.newArrayList(
+                findTopicRequestsByExample(null, teamSelected, null, null, tenantId, null));
+      }
     }
 
-    topicRequests.forEach(
-        topicRequest ->
-            topicRequest.setRequesttimestring(
-                new SimpleDateFormat("dd-MMM-yyyy HH:mm:ss")
-                    .format((topicRequest.getRequesttime()).getTime())));
+    boolean wildcardFilter = (allReqs && wildcardSearch != null && !wildcardSearch.isEmpty());
```

Comment:
why only for allReqs ? can we also apply on 'myteam requests' ?

Reply:
allReqs is for approvers view, as the wildcardSearch is not available on the others, I didn't want to add the extra (albeit very small) process to the other my team requests.